src/context/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React, {
+  useState, useEffect, useMemo,
+} from 'react';
+import { enqueueSnackbar } from 'notistack';
+import { AxiosError } from 'axios';
+import { axiosInstance } from '../utils/apis';
+import { AppContextProps, UserData } from './types';
+import userDataContext from './contextData';
+
+const AppContext: React.FC<AppContextProps> = ({ children }) => {
+  const [userData, setUserData] = useState<UserData[]>([]);

```

Comment:
while userData is an object, why to give it initial value as an empty array, instead it can be null, more meaningfull